File: src/switch_box.rs
use uuid::Uuid;

pub struct SwitchBox {
    pub pin_count: i32,
    pub pins: Vec<Uuid>,
}

impl SwitchBox {

    pub fn new(pin_count: i32) -> Self {
        let mut pins = Vec::new();
        for _i in 0..pin_count {
            pins.push(Uuid::new_v4());
        }
        SwitchBox { pin_count, pins }        
    }

    pub fn get_pins(&self) -> Vec<Uuid> {
        self.pins.clone()
    }

}

File: src/vpga_graph/switch_box.clj
(ns vpga-graph.switch-box 
  (:require [vpga-graph.util :as util]))

(defrecord SwitchBox [pin-count pin-ids])

(defn pin-ids
  "Returns the ids of the Switch Box's pins"
  [switch-box]
  (:pin-ids switch-box))

(defn generate
  "Generates a new Switch Box of the given pin count"
  [pin-count]
  (let [pin-ids (vec (take pin-count (repeatedly util/->UUID)))]
    (SwitchBox. pin-count pin-ids)))